File: core/src/com/broll/gainea/server/core/cards/impl/play/C_TeleportSwamp.java
package com.broll.gainea.server.core.cards.impl.play;

import com.broll.gainea.server.core.cards.AbstractTeleportCard;
import com.broll.gainea.server.core.map.AreaType;
import com.broll.gainea.server.core.map.Location;
import com.broll.gainea.server.core.utils.LocationUtils;

import java.util.List;
import java.util.stream.Collectors;

public class C_TeleportSwamp extends AbstractTeleportCard {
    public C_TeleportSwamp() {
        super(45, "Sumpferkundung", "Bewegt eine Truppe zu einem beliebigen Sumpf auf der gleichen Karte");
    }

    @Override
    public List<Location> getTeleportTargets(Location from) {
        return LocationUtils.filterByType(from.getContainer().getExpansion().getAllLocations(), AreaType.BOG).collect(Collectors.toList());
    }
}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_TeleportSwamp.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.Game
import com.broll.gainea.server.core.cards.TeleportCard
import com.broll.gainea.server.core.map.AreaType.BOG
import com.broll.gainea.server.core.map.Location

class C_TeleportSwamp : TeleportCard(45, "Sumpferkundung", "Bewegt eine Truppe zu einem beliebigen Sumpf auf der gleichen Karte") {
    override fun getTeleportTargets(from: Location) = from.container.expansion.allAreas.filter { it.type == BOG }

    override fun validFor(game: Game) = game.map.allAreas.any { it.type == BOG }
}
